go/enclave/db/rawdb/accessors_receipts.go

```diff
@@ -2,18 +2,24 @@ package rawdb
 
 import (
 	"bytes"
+	"errors"
 	"fmt"
 	"math/big"
-
-	common2 "github.com/obscuronet/go-obscuro/go/common"
+	"sync"
 
 	"github.com/ethereum/go-ethereum/common"
 	"github.com/ethereum/go-ethereum/core/types"
 	"github.com/ethereum/go-ethereum/ethdb"
 	"github.com/ethereum/go-ethereum/params"
 	"github.com/ethereum/go-ethereum/rlp"
+	"github.com/obscuronet/go-obscuro/go/common/errutil"
+
+	common2 "github.com/obscuronet/go-obscuro/go/common"
 )
 
+// make sure that reading and incrementing is done atomically
+var incrementSync sync.Mutex
```

Comment:
Feels a bit weird for this mutex to be a static package var. If there were multiple enclaves running in a sim then this would lock across all of them... Maybe that's fine but thought I'd raise it.